solution/index.js

```diff
@@ -0,0 +1,317 @@
+        data[PropObj].unshift(newTask)
+    }
+    localStorage.setItem("tasks",JSON.stringify(data))
+}
+
+
+// edit task function ---> edit the task if i dblclick
+function editTask (event){
+    const liEl = event.target;
+    liEl.style.backgroundColor = "whitesmoke";                // if i edit the background is change
+    const oldValue = liEl.textContent;                        // remove the value data letter
+    
+    liEl.addEventListener("blur", Blur);
+    function Blur (){
+        const newValue = liEl.textContent;                    // add the correct value to data
+        const section = liEl.closest("section");
+        if (section === null) return;
+        let PropObj = section.id;
+        
+        if (liEl.textContent === ""){                         // if text is empty ---> remove and alert
+            alert("you delete your task");
+            liEl.remove(); 
+            let index = data[PropObj].indexOf(oldValue);
+            data[PropObj].splice(index,1);
+        }
+        else{                                                 // else ---> uptade the task in data
+            let index = data[PropObj].indexOf(oldValue);
+            data[PropObj].splice(index,1,newValue);
+            console.log(data)
+        }
+        liEl.style.backgroundColor="rgb(192, 236, 172)";      // change the background agian
+        localStorage.setItem("tasks",JSON.stringify(data))    // change the local storage to data
+    }
+}
+
+// function switchTasks ---> switchTasks
+let liEl;
+function switchTasks(e){
+    liEl = e.target;
+    document.onkeydown= e=>{
+        if(liEl !== undefined){   
+        const liValue = liEl.textContent;                           // value of li
+        const section = liEl.closest("section");
+        let PropObj = section.id;                                   // for the data
+        
+        if (e.altKey && e.key === '1'){
+            let ul = document.querySelector(".to-do-tasks");        
+            ul.insertBefore(liEl,ul.firstChild);                    // push to dom
+            data["todo"].unshift(liValue);                          // push to first in data
+            let index = data[PropObj].indexOf(liValue);             // get the index
+            data[PropObj].splice(index,1);                          // remove from data
+        }
+        else if (e.altKey && e.key === '2'){
+            let ul = document.querySelector(".in-progress-tasks");       
+            ul.insertBefore(liEl,ul.firstChild);
+            data["in-progress"].unshift(liValue);                      
+            let index = data[PropObj].indexOf(liValue);
+            data[PropObj].splice(index,1);
+        }
+        else if (e.altKey && e.key === '3'){
+            let ul = document.querySelector(".done-tasks");
+            ul.insertBefore(liEl,ul.firstChild);
+            data["done"].unshift(liValue);
+            let index = data[PropObj].indexOf(liValue);
+            data[PropObj].splice(index,1);
```

Comment:
AAAAAAHHH COPY PASTED CODE!!! MY EYESSSS!!!! NOOOO!!!!

Reply:
Haha lol 
I need to update data 
And it's only 3 if 
I don't to create function for this